src/internal/Subscriber.ts

```diff
@@ -163,6 +168,39 @@ export class Subscriber<T> extends Subscription implements Observer<T> {
     this._parents = _parents;
     return this;
   }
+
+  /**
+   * @ignore
+   * @internal
+   * Synchronous errors thrown within `subscribe` and caught by `_trySubscribe`
+   * can only be reported using a subscriber's `error` method if the subscriber
+   * is not stopped. This method traverses the `destination` chain of
+   * subscribers and uses the appropriate unhandled mechanism to report the
+   * error if a stopped subscriber is found.
+   */
+  [Symbol_reportError](err?: any): void {
+    let observer: PartialObserver<T> = this;
+    while (observer) {
+      // `observer` will either be a subscriber or an empty observer, but for
+      // the reasons outlined in the constructor, `instanceof Subscriber`
+      // cannot be used.
+      if (isTrustedSubscriber(observer)) {
+        const { destination, isStopped } = observer;
+        if (isStopped) {
+          if (config.useDeprecatedSynchronousErrorHandling) {
+            throw err;
+          } else {
+            hostReportError(err);
+            return;
+          }
+        }
+        observer = destination;
+      } else {
+        this.error(err);
+        return;
+      }
```

Comment:
it looks like this loop will continue to hit hostReportError over and over for every level at which a subscriber `isStopped`. So if it has this.destination.destination.destination and all are stopped, the same error will be reported 3 or 4 times.

Also, I'm trying to understand the bug this is solving. The `isStopped` flag is meant to guard against code calling `subscriber.complete(); subscriber.error(new Error());` or calling `error` twice, etc.

It seems like the real fix might be to figure out why the Subscriber is prematurely stopped in the first place. Sorry, I should have caught that in an earlier review. Perhaps I missed something in one of your earlier comments?

Reply:
> Sorry, I should have caught that in an earlier review.

No worries. With PRs like this - where there isn't a prior, agreed-upon solution that was discussed in an issue - I'm aware that it could all be for nought. It could be the wrong approach or could just be plain wrong, etc. I probably should have mentioned that at the outset, as it's never fun having to tell someone you don't want to merge a PR.

Anyway, this PR is far from urgent.

> Also, I'm trying to understand the bug this is solving.

The swallowing of the error is problem that it solves. If an error is thrown after a subscriber becomes stopped - as in #3444 - `_trySubscribe` attempts to report it via which `error` is pointless. The motivation for the PR was the appearance of the bug in #3444 in 5.5.6 and its disappearance in the beta. I thought that if a change to the sync error handling was made in 5.5.6 - to better report errors - it would be preferrable it the beta didn't go back to swallowing them.

I guess swallowed errors are a pet peeve of mine.

> It seems like the real fix might be to figure out why the Subscriber is prematurely stopped in the first place.

In #3444, the reason the subscriber became stopped was due to `notifyComplete` being called by the inner subscriber. The `notifyComplete` implementation called `super.complete()` and stopped the subscriber. However, with a synchronous notifier there was more code that ran after that and there was a bug in that code.

It might be possible to re-arrange `repeatWhen` so that becoming stopped is guaranteed to be the very last thing that occurs, but situations in which errors are swallowed could re-arise if someone were to write a wierd, user-land repeat-ish operator.

I imagine that the types of errors that this PR would expose - that would otherwise be swallowed - are very rare. I doubt there are too many operators like `repeatWhen` that have to fudge with the forwarding of `complete` notifications.

> it looks like this loop will continue to hit `hostReportError` over and over for every level at which a subscriber `isStopped`

I don't believe that's the case; the `return` will exit the function and, as it doesn't re-throw, there's no error to be caught by any outer `subscribe` calls. Basically, there are three ways out of the loop:

* the old, sync error behaviour of re-throwing the error;
* calling `hostReportError` and returning;
* calling `error` and returning.

Of those three, I'm quite sure that the last two are correct. However, I'm less certain about the first: the `throw`. I think it's okay, but I will need to give it some more thought.